includes/Core/Conversion_Tracking/Conversion_Tracking.php

```diff
@@ -131,8 +131,25 @@ function ( Conversion_Events_Provider $active_provider ) {
 			}
 		);
 
-		wp_add_inline_script( GTag::HANDLE, 'window._googlesitekit = window._googlesitekit || {};' );
-		wp_add_inline_script( GTag::HANDLE, 'window._googlesitekit.gtagEvent = (name, data) => gtag("event", name, {...data, event_source: "site-kit" });' );
+		$gtag_event = '
+			window._googlesitekit = window._googlesitekit || {};
+			window._googlesitekit.throttledEvents = [];
+			window._googlesitekit.gtagEvent = (name, data) => {
+				var key = JSON.stringify( { name, data } );
+
+				if ( !!window._googlesitekit.throttledEvents[key] ) {
+					return;
+				}
+				window._googlesitekit.throttledEvents[key] = true;
+				setTimeout( () => {
+					delete window._googlesitekit.throttledEvents[key];
+				}, 500 );
+
+				gtag("event", name, {...data, event_source: "site-kit" });
```

Comment:
Super nitpick, but:

```suggestion
				gtag( "event", name, { ...data, event_source: "site-kit" } );
```

😅